src/lib/cloud-manager-hoc.jsx

```diff
@@ -0,0 +1,98 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import {connect} from 'react-redux';
+
+import VM from 'scratch-vm';
+import CloudProvider from '../lib/cloud-provider';
+
+import {
+    getIsShowingWithId
+} from '../reducers/project-state';
+
+/*
+ * Higher Order Component to manage the connection to the cloud dserver.
+ * @param {React.Component} WrappedComponent component to manage VM events for
+ * @returns {React.Component} connected component with vm events bound to redux
+ */
+const cloudManagerHOC = function (WrappedComponent) {
+    class CloudManager extends React.Component {
+        constructor (props) {
+            super(props);
+            this.cloudProvider = null;
+            this.canUseCloud = false;
+        }
+        componentDidUpdate (prevProps) {
+            // If any of the following are not true, do not open a cloud connection
+            const canUseCloud = this.props.cloudHost && this.props.username && this.props.isShowingWithId;
+
+            if (!canUseCloud) return;
+
+            // TODO add scratcher check
+            this.canUseCloud = canUseCloud;
+
+            if ((this.props.username !== prevProps.username) ||
+                (this.props.projectId !== prevProps.projectId) ||
+                (!this.cloudProvider) ||
+                (this.cloudProvider && !this.cloudProvider.connection)) {
+                // TODO need to add provisions for viewing someone
```

Comment:
Because this is in `componentDidUpdate`, it will not be called for the initial render. You probably want to extract lines 35-44 into a separate function that you can call here (during the update) and also in `componentDidMount`. I'm guessing this went unnoticed because the load process would never init the cloud server on mount because the project isn't loaded yet.